xml_converter/integration_tests/run_tests.py

```diff
@@ -166,17 +166,24 @@ def main() -> bool:
             if not re.match(args.filter, testcase.name):
                 continue
 
-        xml_output_dir_path = os.path.join(output_parent_dirpath, "xml", testcase.name)
-        proto_output_dir_path = os.path.join(output_parent_dirpath, "proto", testcase.name)
+        output_xml_paths: Optional[List[str]] = None
+        output_proto_paths: Optional[List[str]] = None
 
-        os.makedirs(xml_output_dir_path, exist_ok=True)
-        os.makedirs(proto_output_dir_path, exist_ok=True)
+        if testcase.expected_output_xml_path:
+            xml_output_dir_path = os.path.join(output_parent_dirpath, "xml", testcase.name)
+            os.makedirs(xml_output_dir_path, exist_ok=True)
+            output_xml_paths = [xml_output_dir_path]
+
+        if testcase.expected_output_proto_path:
+            proto_output_dir_path = os.path.join(output_parent_dirpath, "proto", testcase.name)
+            os.makedirs(proto_output_dir_path, exist_ok=True)
+            output_proto_paths = [proto_output_dir_path]
 
         rawstdout, rawstderr, returncode = run_xml_converter(
             input_xml=testcase.xml_input_paths,
             input_proto=testcase.proto_input_paths,
-            output_xml=[xml_output_dir_path],
-            output_proto=[proto_output_dir_path],
+            output_xml=output_xml_paths if output_xml_paths else None,
+            output_proto=output_proto_paths if output_proto_paths else None,
```

Comment:
What is the difference here between passing in an empty list and passing in `None`? These lines seem to want to special case an `[]` empty list as the same thing as `None`.